src/coreComponents/mesh/CellElementSubRegion.cpp

```diff
@@ -42,24 +47,71 @@ CellElementSubRegion::CellElementSubRegion( string const & name, Group * const p
 
 CellElementSubRegion::~CellElementSubRegion()
 {
-  // TODO Auto-generated destructor stub
+  // Left blank
 }
 
-void CellElementSubRegion::copyFromCellBlock( CellBlock & source )
+void CellElementSubRegion::setElementType( ElementType elementType )
 {
-  this->setElementType( source.getElementType());
-  this->setNumNodesPerElement( source.numNodesPerElement() );
-  this->setNumFacesPerElement( source.numFacesPerElement() );
-  this->resize( source.size());
-  this->nodeList() = source.nodeList();
-
-  arrayView1d< globalIndex const > const sourceLocalToGlobal = source.localToGlobalMap();
-  this->m_localToGlobalMap.resize( sourceLocalToGlobal.size() );
-  for( localIndex i = 0; i < localToGlobalMap().size(); ++i )
+  m_elementType = elementType;
+
+  switch( m_elementType )
   {
-    this->m_localToGlobalMap[ i ] = sourceLocalToGlobal[ i ];
+    case ElementType::Hexahedron:
+    {
+      m_numNodesPerElement = 8;
+      m_numEdgesPerElement = 12;
+      m_numFacesPerElement = 6;
+      break;
+    }
+    case ElementType::Tetrahedron:
+    {
+      // Tetrahedron
+      m_numNodesPerElement = 4;
+      m_numEdgesPerElement = 6;
+      m_numFacesPerElement = 4;
+      break;
+    }
+    case ElementType::Prism:
+    {
+      // Triangular prism
+      m_numNodesPerElement = 6;
+      m_numEdgesPerElement = 9;
+      m_numFacesPerElement = 5;
+      break;
+    }
+    case ElementType::Pyramid:
+    {
+      // Pyramid
+      m_numNodesPerElement = 5;
+      m_numEdgesPerElement = 8;
+      m_numFacesPerElement = 5;
+      break;
+    }
+    default:
+    {
+      GEOSX_ERROR( "Error.  Don't know what kind of element this is." );
```

Comment:
Do we want to have a more explicit error message, with the name of the problematic subregion, etc?

Reply:
I was wrong, since our `ElementType` does not distinguish between 1, 2 and 3 dimensional elements (for edges, faces, cells). It's therefore legit not to consider all the cases in this switch.
Maybe splitting `ElementType` to consider those dimensions could be a move.